### ISSUE TITLE
Index 0 out of bounds for length 0

### ISSUE THREAD

**Issue by yooo1999** [NationalSecurityAgency/ghidra#6264]
```
Index 0 out of bounds for length 0
java.lang.IndexOutOfBoundsException: Index 0 out of bounds for length 0
	at java.base/jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:64)
	at java.base/jdk.internal.util.Preconditions.outOfBoundsCheckIndex(Preconditions.java:70)
	at java.base/jdk.internal.util.Preconditions.checkIndex(Preconditions.java:266)
	at java.base/java.util.Objects.checkIndex(Objects.java:359)
	at java.base/java.util.ArrayList.get(ArrayList.java:427)
	at ghidra.program.model.lang.protorules.HomogeneousAggregate.filter(HomogeneousAggregate.java:74)
	at ghidra.program.model.lang.protorules.ModelRule.assignAddress(ModelRule.java:162)
	at ghidra.program.model.lang.ParamListStandard.assignAddress(ParamListStandard.java:134)
	at ghidra.program.model.lang.ParamListStandard.assignMap(ParamListStandard.java:181)
	at ghidra.program.model.lang.PrototypeModel.assignParameterStorage(PrototypeModel.java:319)
	at ghidra.program.model.lang.PrototypeModel.getStorageLocations(PrototypeModel.java:363)
	at ghidra.program.database.function.FunctionDB.updateParametersAndReturn(FunctionDB.java:830)
	at ghidra.program.database.function.FunctionDB.loadVariables(FunctionDB.java:723)
	at ghidra.program.database.function.FunctionDB.getVariable(FunctionDB.java:1873)
	at ghidra.program.database.symbol.VariableSymbolDB.getObject(VariableSymbolDB.java:156)
	at ghidra.program.database.symbol.NamespaceManager.getAddressSet(NamespaceManager.java:275)
	at ghidra.program.database.function.FunctionDB.getBody(FunctionDB.java:345)
	at ghidra.app.cmd.analysis.SharedReturnAnalysisCmd.checkBelowFunction(SharedReturnAnalysisCmd.java:318)
	at ghidra.app.cmd.analysis.SharedReturnAnalysisCmd.applyTo(SharedReturnAnalysisCmd.java:88)
	at ghidra.framework.cmd.BackgroundCommand.applyTo(BackgroundCommand.java:51)
	at ghidra.app.plugin.core.function.SharedReturnAnalyzer.added(SharedReturnAnalyzer.java:79)
	at ghidra.app.plugin.core.analysis.AnalysisScheduler.runAnalyzer(AnalysisScheduler.java:186)
	at ghidra.app.plugin.core.analysis.AnalysisTask.applyTo(AnalysisTask.java:39)
	at ghidra.app.plugin.core.analysis.AutoAnalysisManager$AnalysisTaskWrapper.run(AutoAnalysisManager.java:686)
	at ghidra.app.plugin.core.analysis.AutoAnalysisManager.startAnalysis(AutoAnalysisManager.java:786)
	at ghidra.app.plugin.core.analysis.AutoAnalysisManager.startAnalysis(AutoAnalysisManager.java:665)
	at ghidra.app.plugin.core.analysis.AutoAnalysisManager.startAnalysis(AutoAnalysisManager.java:630)
	at ghidra.app.plugin.core.analysis.AnalysisBackgroundCommand.applyTo(AnalysisBackgroundCommand.java:58)
	at ghidra.framework.plugintool.mgr.BackgroundCommandTask.run(BackgroundCommandTask.java:103)
	at ghidra.framework.plugintool.mgr.ToolTaskManager.run(ToolTaskManager.java:334)
	at java.base/java.lang.Thread.run(Thread.java:833)

---------------------------------------------------
Build Date: 2023-Dec-22 0936 EST
Ghidra Version: 11.0
Java Home: C:\Program Files\Java\jdk-17.0.1
JVM Version: Oracle Corporation 17.0.1
OS: Windows 10 10.0 amd64
Workstation: myComp

```

**Comment by ryanmkurtz** [comment]
Was this an aarch64 binary?

**Comment by yooo1999** [comment]
@ryanmkurtz Yes, aarch64 binary

**Comment by ryanmkurtz** [comment]
This should have been fixed in Ghidra 11.0.1.
